File: src/org/home/advent2022/P1.java
package org.home.advent2022;

import java.util.PriorityQueue;
import java.util.Scanner;

public class P1 {

    public static void main(String[] args) {
        final Scanner input = new Scanner(System.in);

        long current = 0;

        final PriorityQueue<Long> pq = new PriorityQueue<>(4);

        while (input.hasNext()) {
            try {
                current += Long.valueOf(input.nextLine());
            } catch (NumberFormatException e) {
                pq.add(current);

                if(pq.size() > 3) {
                    pq.poll();
                }

                current = 0;
            }
        }

        System.out.println(pq.stream().mapToLong(it -> it).sum());
    }
}

File: src/org/home/kickstart/roundk/P1.kt
package org.home.kickstart.roundk

import java.util.*

fun main(args: Array<String>) {
    val input = Scanner(System.`in`)

    val t = input.nextLine().toInt();

    for (i in 1..t) {
        val n = input.nextLine().toInt()
        val b = input.nextLine()!!.map { it == '1' }.toBooleanArray()

        println("Case #$i: " + solve(n, b))
    }
}

private fun solve(n: Int, b: BooleanArray): Long {

    val l = LongArray(b.size)
    var r = LongArray(b.size)

    var k = -1L
    for (i in 0 until b.size) {
        if (b[i]) k = i.toLong()

        if (k < 0) l[i] = n.toLong()
        else l[i] = i - k
    }

    k = -1L
    for (i in b.size - 1 downTo 0) {
        if (b[i]) k = i.toLong()

        if (k < 0) r[i] = n.toLong()
        else r[i] = k - i
    }

    return l.zip(r).map { minOf(it.first, it.second) }.sum()
}
